### ISSUE TITLE
CFG does not correctly support try-catch blocks with unconditional return insude

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-csharp#621]
The CFG implementation in the SonarC# uses approximation of the possible execution paths and does not correctly handle try-catch statements with unconditional `return` inside the `try` block. This leads to false positives in the rules that are based on the CFG.

For example, the following code will generate false positive in S2583 (Conditionally executed blocks should be reachable):
```
object o;
try
{
    o = new object();
    return;
}
catch
{
    if (o != null) // issue raised here
        DoSomething();
}
```
